functions/src/requests/index.ts

```diff
@@ -101,35 +104,35 @@ export const triggerEventsWhenRequestIsCreated = functions.firestore
   // eslint-disable-next-line no-unused-vars,@typescript-eslint/no-unused-vars
   .onCreate((snapshot: DocumentSnapshot, context: EventContext) => {
     return validateRequest(snapshot.data() as IRequest)
+      .then(() => {
+        return indexRequest(snapshot);
+      })
       .catch(errors => {
         console.error('Invalid Request Found: ', errors);
         return firestore()
           .collection('requests')
           .doc(context.params.requestId)
           .delete();
-      })
-      .then(() => {
-        return indexRequest(snapshot);
```

Comment:
Same issue here, the request would be indexed even if the trigger had deleted it.